### ISSUE TITLE
[Menu] Accessibility issues for sp-menu & sp-menu-item

### ISSUE THREAD

**Issue by stanvladut** [adobe/spectrum-web-components#1508]
### Expected Behaviour
The `sp-menu` & `sp-menu-item` combination should be a fully accessible combination, having the roles `menu` and `menuitem`, respectively.

### Actual Behaviour
The roles are actually applied for these two components excepting the case when the `href` property is passed to the `sp-menu-item` component. If the `sp-menu-item` components are configured to have the `href` property set, then our tests checking the component for being fully accessible fail with this error:
```
 Error: Accessibility Violations
      ---
      Rule: aria-required-children
      Impact: critical
      Certain ARIA roles must contain particular children (https://dequeuniversity.com/rules/axe/4.1/aria-required-children?application=axeAPI)
      
      Issue target: sp-menu
      Context: <sp-menu role="menu" dir="ltr" tabindex="0">
      Fix any of the following:
        Required ARIA children role not present: menuitemradio, menuitem, menuitemcheckbox
```

#### Steps to Reproduce
Test this structure for accessibility:
```
<sp-menu>
  <sp-menu-item href="#">Deselect</sp-menu-item>
  <sp-menu-item href="#">Select Inverse</sp-menu-item>
  <sp-menu-item href="#">Feather...</sp-menu-item>
  <sp-menu-item href="#">Select and Mask...</sp-menu-item>
  <sp-menu-divider></sp-menu-divider>
  <sp-menu-item href="#">Save Selection</sp-menu-item>
  <sp-menu-item href="#" disabled>Make Work Path</sp-menu-item>
</sp-menu>
```
